### PR TITLE
Adapt BG selector according to design session

### PR DESCRIPTION
**Pull request by cedricmoullet**
Desktop:
Background text
White separator
Remove button after click
Mobile:
Put the button bottom-left
Open to top
Keep same icon

### CODE REVIEW

**Comment by cedricmoullet** [comment]
Test link: https://mf-geoadmin3.dev.bgdi.ch/moc_fixbgselector/
@davidoesch please test

**Comment by davidoesch** [comment]
- mobile ok ( to be honest: icons as well now on the right left side - the app is less compact --- I get the visual feeling that there are more icons, cluttered all over the interface )

- desktop
A) arrow: use the same one as for the topic fly out ( consistency)

B) responsive design: only for desktop the rectangle button - for all other states the round icon: all those iframes would be fully covered with the rectangle 


Conclusio:
I would keep the Mobile and all other states except desktop as it is currently.: the map is tidied up : only the top left contains the control. And the user iframe embedder has the FULL left part of the map to show his data. 

Happy to discuss this over phone

**Comment by cedricmoullet** [comment]
The reason why the mobile icon has been moved is because we wanted to
define "mental zones":
- top right: navigation/position
- bottom right: storage/share
- bottom left: content
So, we have here a clear case where esthetic doesn't fit with information
architecture.
We have to define if we want to promote esthetic over usability.

I don't get your point regarding responsiveness and embed: the round icon
is used everywhere (mobile, embed) except desktop. Is it not what you want
?

Cédric

2015-06-05 6:51 GMT+02:00 davidoesch <notifications@github.com>:

>
>    -
>
>    mobile ok ( to be honest: icons as well now on the right left side -
>    the app is less compact --- I get the visual feeling that there are more
>    icons, cluttered all over the interface )
>    -
>
>    desktop
>    A) arrow: use the same one as for the topic fly out ( consistency)
>
> B) responsive design: only for desktop the rectangle button - for all
> other states the round icon: all those iframes would be fully covered with
> the rectangle
>
> Conclusio:
> I would keep the Mobile and all other states except desktop as it is
> currently.: the map is tidied up : only the top left contains the control.
> And the user iframe embedder has the FULL left part of the map to show his
> data.
>
> Happy to discuss this over phone
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2397#issuecomment-109160548>
> .
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

**Comment by cedricmoullet** [comment]
One important: the mental zoning concept allows to promote the bg selector icon. I believe it is more "discoverable" if it's on the bottom left.

**Comment by davidoesch** [comment]
A) "mental zoning concept" I agree it is more discoverable better for promoting. My impression is: we overestimated the need for switching the BG, it is done usually once, then not again. Especially if you embed an iframe: you as a "embedder" decide which bg fits your purpose: you set it. while embedded you want as much data as possible
for me is the question : show as much space for data or follow usability ( I do tend for data)
-> for mobile put it where it is currently

space


b) responsiveness: in desktop mode , on smaller screens

state 1: ok for rectangle



state 2:  use the circle (where it is today or LR) instead of the rectangle 


state 3: use the circle instead of the rectangle (where it is today or LR)



ceterum censeo: usability or give as much as possible space for maps.... I clearly tend for space for maps ( as does gmaps, bing,search.ch) ....

**Comment by cedricmoullet** [comment]
Responsiveness: no. The difference is made between the mode: mobile, embed and desktop. It's a non sense to replace the rectangle version by the round version if the user changes the window size within desktop mode. Maybe a smaller rectangle would make sense, but for sure not the round version on the left.

**Comment by cedricmoullet** [comment]
For the rest: I don't see where we loose space on mobile. Not sure that the green/red schema is correct. 
And it was not the question of the importance of the BG selector but simply the fact that it was not discovered at all.

**Comment by davidoesch** [comment]
responsivness: for state 2 and state 3 we are just using up precious map space --- the one we gained for the user when we ditched the overview map... now it will be reclaimed by bg selector: for sth you need *once* when using the map..... agree, changing from square to circel might make no sense...a smaller square would do

embed: to much space taken anyway.... how does google solve this? --very small toggle

**Comment by cedricmoullet** [comment]
i would remove it on embed

2015-06-05 8:18 GMT+02:00 davidoesch <notifications@github.com>:

> responsivness: for state 2 and state 3 we are just using up precious map
> space --- the one we gained for the user when we ditched the overview
> map... now it will be reclaimed by bg selector: for sth you need *once*
> when using the map..... agree, changing from square to circel might make no
> sense...a smaller square would do
>
> embed: to much space taken anyway.... how does google solve this? --very
> small toggle
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2397#issuecomment-109173495>
> .
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

**Comment by davidoesch** [comment]
+1 for removing it on embed

**Comment by cedricmoullet** [comment]
Embed removed. 
Small rectangle for those having huge screens but trying to see a mini swiss geoportal on desktop.
Test link updated.

**Comment by oterral** [comment]
The height of the close button should be the same as the height of bg layers
elements.
On small screen we don't see the "No bg layer" element when we are outside the switzerland.

**Comment by cedricmoullet** [comment]
for the height, we'll see that on monday....

2015-06-05 9:03 GMT+02:00 oterral <notifications@github.com>:

> The height of the close button should be the same as the height of bg
> layers
> elements.
> On small screen we don't see the "No bg layer" element when we are outside
> the switzerland.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2397#issuecomment-109182312>
> .
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

**Comment by davidoesch** [comment]
thanks @cedricmoullet : i like the changes 

"On small screen we don't see the "No bg layer" element when we are outside the switzerland."
-> valid point, and this is why went initally for the grey border of the selctor. IMHO: we have no to live with that outside of switzerland issue, or use a grey border inside the image or make the void layer image darker

"height"-> have look on monday

**Comment by cedricmoullet** [comment]
80/20

2015-06-05 9:10 GMT+02:00 davidoesch <notifications@github.com>:

> thanks @cedricmoullet <https://github.com/cedricmoullet> : i like the
> changes
>
> "On small screen we don't see the "No bg layer" element when we are
> outside the switzerland."
> -> valid point, and this is why went initally for the grey border of the
> selctor. IMHO: we have no to live with that outside of switzerland issue,
> or use a grey border inside the image or make the void layer image darker
>
> "height"-> have look on monday
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2397#issuecomment-109184489>
> .
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

**Comment by oterral** [comment]
for the color white/black we could try to add 2 borders one black and one white.

bg-2colors

**Comment by cedricmoullet** [comment]
someone is dead ?

2015-06-05 9:13 GMT+02:00 oterral <notifications@github.com>:

> for the color white/black we could try to add 2 borders one black and one
> white.
>
> [image: bg-2colors]
> <https://cloud.githubusercontent.com/assets/621420/8001149/142e4488-0b63-11e5-810e-c2dacf54919a.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2397#issuecomment-109184732>
> .
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

**Comment by cedricmoullet** [comment]
@oterral @gjn Ready for review and merge

**Comment by pfanguin** [comment]
When the window is more little, the size of rectangles is more little, but not the arrow. 
capture d ecran 2015-06-08 a 16 35 37

File: src/index.mako.html [hunk 1]
```diff
@@ -233,6 +230,14 @@
     </div>
 % endif
 
+% if device == 'mobile':
+    <div ga-background-selector ga-background-selector-map="map" class="ga-bg-mobile" ng-show="!globals.offline"></div>
+% endif
+
+% if device == 'embed':
+    <div ga-background-selector ga-background-selector-map="map"></div>
```
Comment (oterral) [inline src/index.mako.html]: Where is the bg selector on embed page ? I don't see it

**Comment by oterral** [comment]
I have 2 suggestions:
  - Put the bg color of the text "Fond de carte in black with opacity 0.7, otherwise we don't see what is written
  - Maybe center the text of bg layer

File: src/index.mako.html (hunk 1 above)
Comment (cedricmoullet) [inline src/index.mako.html]: I removed it.

2015-06-09 12:50 GMT+02:00 oterral <notifications@github.com>:

> In src/index.mako.html
> <https://github.com/geoadmin/mf-geoadmin3/pull/2397#discussion_r32001193>:
>
> > @@ -233,6 +230,14 @@
> >      </div>
> >  % endif
> >
> > +% if device == 'mobile':
> > +    <div ga-background-selector ga-background-selector-map="map" class="ga-bg-mobile" ng-show="!globals.offline"></div>
> > +% endif
> > +
> > +% if device == 'embed':
> > +    <div ga-background-selector ga-background-selector-map="map"></div>
>
> Where is the bg selector on embed page ? I don't see it
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2397/files#r32001193>.
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

**Comment by cedricmoullet** [comment]
Ok, please patch

2015-06-09 12:51 GMT+02:00 oterral <notifications@github.com>:

> I have 2 suggestions:
>
>    - Put the bg color of the text "Fond de carte in black with opacity
>    0.7, otherwise we don't see what is written
>    - Maybe center the text of bg layer
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geoadmin/mf-geoadmin3/pull/2397#issuecomment-110317212>
> .
>



-- 
Twitter: http://twitter.com/cedricmoullet
Linked In: http://www.linkedin.com/in/cedricmoullet

<http://twitter.com/cedricmoullet>

**Comment by oterral** [comment]
@davidoesch you validate the black bg on desktop ? https://mf-geoadmin3.dev.bgdi.ch/moc_fixbgselector/

**Comment by oterral** [comment]
For me it's good to merge

**Comment by cedricmoullet** [comment]
jenkins test

**Comment by cedricmoullet** [comment]
@davidoesch could you please check oli's changes here: https://mf-geoadmin3.dev.bgdi.ch/moc_fixbgselector/

**Comment by davidoesch** [comment]
ok for merge --- nice job @cedricmoullet @oterral

**Pull request merged**